flycheck.el

```diff
@@ -1249,6 +1273,26 @@ FILE-NAME is nil, return `default-directory'."
 (defvar read-flycheck-checker-history nil
   "`completing-read' history of `read-flycheck-checker'.")
 
+(defun flycheck-completing-read (prompt candidates default &optional history)
```

Comment:
Thanks for extracting this one.  It's a first step, and I think it needs further cleanup.  Not part of this PR, of course, but it's something that I'll need to remember.  If you want, you can drop me a `;; TODO` comment here :)